data_diff/databases/snowflake.py

```diff
@@ -35,10 +37,21 @@ def __init__(self, *, schema: str, **kw):
         logging.getLogger("snowflake.connector.network").disabled = True
 
         assert '"' not in schema, "Schema name should not contain quotes!"
-        self._conn = snowflake.connector.connect(
-            schema=f'"{schema}"',
-            **kw,
-        )
+        if (
+            "password" not in kw and "key" in kw
+        ):  # if private keys are used instead of password for Snowflake connection, read in key from path specified and pass as "private_key" to connector.
+            with open(kw.get("key"), "rb") as key:
+                p_key = serialization.load_pem_private_key(key.read(), password=None, backend=default_backend())
```

Comment:
It is common to use encrypted private keys, i.e. the ones with a password. In that case, both `password` & `key` will be present, and the password should be passed to the PEM loader.

Can you please slightly adjust the logic here to check only for the `key` presence (and no `password` absence), and pass down the key to the PEM loader instead of the connection (`kw.pop()` perhaps)?

If it is problematic, we can do it later as a separate PR.